client_test/integration_test.go

```diff
@@ -0,0 +1,178 @@
+			}
+			infra := setupSharedInra(tc)
+
+			// Setup clients
+			clientA := setupIntegrationClient(tc, testactors.AliceName, infra)
+			clientB := setupIntegrationClient(tc, testactors.BobName, infra)
+
+			intermediaries := []client.Client{setupIntegrationClient(tc, testactors.IreneName, infra)}
+			intermediaryAddresses := []types.Address{*intermediaries[0].Address}
+			if tc.NumOfHops == 2 {
+				intermediaries = append(intermediaries, setupIntegrationClient(tc, testactors.BrianName, infra))
+				intermediaryAddresses = append(intermediaryAddresses, *intermediaries[1].Address)
+			}
+
+			defer clientA.Close()
+			defer clientB.Close()
+			defer func() {
+				for _, clientI := range intermediaries {
+					clientI.Close()
+				}
+			}()
+			asset := common.Address{}
+			// Setup ledger channels between Alice/Bob and intermediaries
+			aliceLedgers := make([]types.Destination, tc.NumOfHops)
+			bobLedgers := make([]types.Destination, tc.NumOfHops)
+			for i, clientI := range intermediaries {
+				// Setup and check the ledger channel between Alice and the intermediary
+				aliceLedgers[i] = setupLedgerChannel(t, clientA, clientI, asset)
+				checkLedgerChannel(t, aliceLedgers[i], initialLedgerOutcome(*clientA.Address, *clientI.Address, asset), query.Ready, &clientA)
+				// Setup and check the ledger channel between Bob and the intermediary
+				bobLedgers[i] = setupLedgerChannel(t, clientI, clientB, asset)
+				checkLedgerChannel(t, bobLedgers[i], initialLedgerOutcome(*clientI.Address, *clientB.Address, asset), query.Ready, &clientB)
+
+			}
+
+			if tc.NumOfHops == 2 {
+				setupLedgerChannel(t, intermediaries[0], intermediaries[1], asset)
+			}
+			// Setup virtual channels
+			objectiveIds := make([]protocols.ObjectiveId, tc.NumOfChannels)
+			virtualIds := make([]types.Destination, tc.NumOfChannels)
+			for i := 0; i < int(tc.NumOfChannels); i++ {
+				outcome := td.Outcomes.Create(testactors.Alice.Address(), testactors.Bob.Address(), virtualChannelDeposit, 0, types.Address{})
+				response := clientA.CreateVirtualPaymentChannel(
+					intermediaryAddresses,
+					testactors.Bob.Address(),
+					0,
+					outcome,
+				)
+				objectiveIds[i] = response.Id
+				virtualIds[i] = response.ChannelId
+
+			}
+			// Wait for all the virtual channels to be ready
+			waitForObjectives(t, clientA, clientB, intermediaries, objectiveIds)
+
+			// Check all the virtual channels
+			for i := 0; i < len(virtualIds); i++ {
+				checkPaymentChannel(t,
+					virtualIds[i],
+					initialPaymentOutcome(*clientA.Address, *clientB.Address, asset),
+					query.Ready,
+					&clientA, &clientB)
+			}
+
+			// Send payments
+			for i := 0; i < len(virtualIds); i++ {
+				for j := 0; j < int(tc.NumOfPayments); j++ {
+					clientA.Pay(virtualIds[i], big.NewInt(int64(1)))
+				}
+			}
+
+			// Wait for all the vouchers to be received by bob
+			for i := 0; i < len(virtualIds); i++ {
+				<-clientB.ReceivedVouchers()
+			}
+
+			// TODO: This check sometime flickers
```

Comment:
Is there an example of this flicker? How does the test fail?

Reply:
This turned out to be a self-caused flicker 🤦 We were waiting for `numVirtualChannel` vouchers. However if we're making more than 1 payment per channel we wouldn't be waiting for all the vouchers. Updated in https://github.com/statechannels/go-nitro/pull/1193/commits/e9ea072f974694ad94d5945f3847e4851cc88f77